src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -229,13 +241,40 @@ private static bool ValidateAzureArcEnvironment(string identityEndpoint, string
             return false;
         }
 
+        /// <summary>
+        /// Creates an MsalClientException for when no managed identity source is available,
+        /// including detailed failure information from IMDS probes if available.
+        /// </summary>
+        private static MsalClientException CreateManagedIdentityUnavailableException(ManagedIdentitySourceResult sourceResult)
+        {
+            string errorMessage = MsalErrorMessage.ManagedIdentityAllSourcesUnavailable;
+
+            if (sourceResult != null)
+            {
+                if (!string.IsNullOrEmpty(sourceResult.ImdsV1FailureReason) || !string.IsNullOrEmpty(sourceResult.ImdsV2FailureReason))
+                {
+                    errorMessage += " IMDS probe failures:";
```

Comment:
`IMDS probe` is a technical concept and ppl using the library won't understand. How about "MSAL was not able to detect the Azure Instance Metadata Service (IMDS) that runs on VMs ..."

Reply:
done